File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> ans;
        unordered_map<int,int> hp;
        for(int i = 0;i < nums.size(); i++){
            int val = target - nums[i];
            auto it = hp.find(nums[i]);
            if(it != hp.end()){
                if(hp[val] != i){
                    ans.push_back(hp[nums[i]]);
                    ans.push_back(i);
                    break;
                }
            }
            hp[val] = i;
        }
        return ans;
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> hp = new HashMap<>();
        for(int i = 0;i < nums.length;i++){
            hp.put(nums[i],i);
        }
        
        int [] ans = new int[2];
        for(int i = 0;i < nums.length ;i++){
            if(hp.containsKey(target-nums[i]) && hp.get(target-nums[i]) != i){
                return new int[] {hp.get(target-nums[i]),i};
            }
        }
        return ans;
    }
}